### PR TITLE
Write search tests

### PR DESCRIPTION
**Pull request by juhyep**
- wrote test cases for #44 in test/categories.js
- one test case checks for if the search keyword is in the title of the post
- second test case checks for if the search keyword is in the description of the post
- I addressed Lara's comment and deleted the dead code and the other files I didn't need. I also edited the last test so that it used a property of 'topic'.

### CODE REVIEW

**Review by adambournes21 — APPROVED** [review]
good job with the test, I can't find any bugs.

**Review by l-ma — CHANGES_REQUESTED** [review]
The files `package 2.json`, `package 3.json`, `pidfile 3`, and `pidfile 4` should not be pushed. I think those should be removed from the PR before merging to main.

Also, there seem to be multiple files with changes for the anonymous feature. If those changes were abandoned, they should be removed before we merge to main. We don't want dead code in the final codebase.

For the tests: the first one LGTM. I don't see how the second one could be passing. A topic does not have a content field, so I don't know what is happening in line 185. Could you please explain it?

**Review by megchityala — COMMENTED** [review]
It seems like there is one error accessing the TopicSearch, but other than that your tests look good.

**Review by megchityala — APPROVED** [review]
Looks good, thanks for fixing the tests!

File: test/categories.js [hunk 1]
```diff
@@ -151,6 +151,42 @@ describe('Categories', () => {
...
+                cid: categoryObj.cid,
+                start: 0,
+                stop: 10,
+                uid: 0,
+                keyword: keyword,
+                sort: 'oldest_to_newest',
+            }, (err, result) => {
+                assert.equal(err, null);
+                assert(Array.isArray(result.topics));
+                assert(result.topics.every(topic => topic instanceof Object && topic.data.includes(keyword)));
```
Comment (l-ma) [inline test/categories.js:185]: topic has no `data` field

**Comment by l-ma** [comment]
Logic Included in #57

**Pull request closed without merging**